src/BundlerMinifier.Core/Bundle/BundleFileProcessor.cs

```diff
@@ -124,35 +124,49 @@ private bool ProcessBundle(string baseFolder, Bundle bundle)
                 }
             }
 
-            string minFile = BundleMinifier.GetMinFileName(bundle.GetAbsoluteOutputFile());
-
-            if (bundle.IsMinificationEnabled || bundle.IsGzipEnabled)
+            MinificationResult minResult = null;
+            var minFile = BundleMinifier.GetMinFileName(bundle.GetAbsoluteOutputFile());
+            if (bundle.IsMinificationEnabled)
             {
                 var outputWriteTime = File.GetLastWriteTimeUtc(minFile);
-                if (!bundle.IsGzipEnabled && bundle.MostRecentWrite < outputWriteTime)
-                    return false;
-                var result = BundleMinifier.MinifyBundle(bundle);
+                var minifyChanged = bundle.MostRecentWrite >= outputWriteTime;
 
-                // If no change is detected, then the minFile is not modified, so we need to update the write time manually
-                if (!result.Changed && File.Exists(minFile))
-                    File.SetLastWriteTimeUtc(minFile, DateTime.UtcNow);
-                changed |= result.Changed;
-
-                if (bundle.IsMinificationEnabled && bundle.SourceMap && !string.IsNullOrEmpty(result.SourceMap))
+                if (minifyChanged)
                 {
-                    string mapFile = minFile + ".map";
-                    bool smChanges = FileHelpers.HasFileContentChanged(mapFile, result.SourceMap);
+                    minResult = BundleMinifier.MinifyBundle(bundle);
+
+                    // If no change is detected, then the minFile is not modified, so we need to update the write time manually
+                    if (!minResult.Changed && File.Exists(minFile))
+                        File.SetLastWriteTimeUtc(minFile, DateTime.UtcNow);
```

Comment:
So just to explain why it is useful for: If you update an input file with a space, the resulting minified is the same as before so `minResult.Changed` will be `false`. Without setting this `SetLastWriteTimeUtc` then the minifier will always run.